student/src/main/java/tds/student/web/StudentContext.java

```diff
@@ -188,20 +193,24 @@ public static String getGrade () {
   }
 
   public static OpportunityInstance getOppInstance () {
-    if (!isAuthenticated ())
+    if (!isAuthenticated ()) {
       return null;
-    UUID oppKey = UUID.fromString (TDSIdentity.getCurrentTDSIdentity ().get ("O_KEY"));
+    }
+
+    TDSIdentity tdsIdentity = TDSIdentity.getCurrentTDSIdentity();
+    UUID oppKey = UUID.fromString (tdsIdentity.get ("O_KEY"));
     UUID sessionKey = UUID.fromString (StudentCookie.getCookieData ("S_KEY"));
     UUID browserKey = UUID.fromString (StudentCookie.getCookieData ("O_BKEY"));
 
-    // check if opp instance exists
-    // if (oppKey == Guid.Empty || sessionKey == Guid.Empty || browserKey ==
-    // Guid.Empty) return null;
-    if (oppKey == null || sessionKey == null || browserKey == null)
-      return null;
+    UUID examBrowserKey = StudentCookie.getCookieData(EXAM_BROWSER_KEY) == null
+      ? null
+      : UUID.fromString(StudentCookie.getCookieData(EXAM_BROWSER_KEY));
+
+    UUID examId = tdsIdentity.get(EXAM_BROWSER_KEY) == null
```

Comment:
Is `EXAM_BROWSER_KEY` supposed to be for both of these?  `examBrowserKey` and `examId` . Maybe it should be `EXAM_ID_COOKIE_KEY` for one of them?